backend/src/api/websocket-handler.ts

```diff
@@ -90,258 +91,266 @@ class WebsocketHandler {
   }
 
   setupConnectionHandling() {
-    if (!this.wss) {
-      throw new Error('WebSocket.Server is not set');
+    if (!this.websocketServers.length) {
+      throw new Error('no websocket server set, cannot setup connection handlers');
     }
 
-    this.wss.on('connection', (client: WebSocket, req) => {
-      this.numConnected++;
-      client['remoteAddress'] = req.headers['x-forwarded-for'] || req.socket?.remoteAddress || 'unknown';
-      client.on('error', (e) => {
-        logger.info(`websocket client error from ${client['remoteAddress']}: ` + (e instanceof Error ? e.message : e));
-        client.close();
-      });
-      client.on('close', () => {
-        this.numDisconnected++;
-      });
-      client.on('message', async (message: string) => {
-        try {
-          const parsedMessage: WebsocketResponse = JSON.parse(message);
-          const response = {};
-
-          const wantNow = {};
-          if (parsedMessage && parsedMessage.action === 'want' && Array.isArray(parsedMessage.data)) {
-            for (const sub of wantable) {
-              const key = `want-${sub}`;
-              const wants = parsedMessage.data.includes(sub);
-              if (wants && client['wants'] && !client[key]) {
-                wantNow[key] = true;
+    this.websocketServers.forEach(wss => {
```

Comment:
This is the actual change. We have an array of websocket servers to setup now. The big diff is due to indentation change.